glnx-fdio.c

```diff
@@ -703,33 +693,26 @@ glnx_loop_write(int fd, const void *buf, size_t nbytes)
   return 0;
 }
 
-/* Read from @fdf until EOF, writing to @fdt.  If @try_reflink is %TRUE,
- * attempt to use any "reflink" functionality; see e.g. https://lwn.net/Articles/331808/
+/* Read from @fdf until EOF, writing to @fdt.  This will use
+ * the generic Linux copy_file_range() syscall if available.
  *
  * The file descriptor @fdf must refer to a regular file.
  *
  * If provided, @max_bytes specifies the maximum number of bytes to read from @fdf.
  * On error, this function returns `-1` and @errno will be set.
  */
 int
-glnx_regfile_copy_bytes (int fdf, int fdt, off_t max_bytes, gboolean try_reflink)
+glnx_regfile_copy_bytes (int fdf, int fdt, off_t max_bytes)
 {
+  /* Last updates from systemd as of commit 6bda23dd6aaba50cf8e3e6024248cf736cc443ca */
+  static int have_cfr = -1; /* -1 means unknown */
+  bool try_cfr = have_cfr != 0;
   bool try_sendfile = true;
```

Comment:
I just noticed now that we're using C99 bools. The future (of 18 years ago) is now! 😎